algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java

```diff
@@ -259,6 +258,28 @@ static EvidenceDirection resolveDirection(@Nullable String direction) {
         return null;
     }
 
+    @NotNull
+    @VisibleForTesting
+    static String extractEvidenceYear(@NotNull LocalDate entryDate, @NotNull List<Reference> references, Therapy therapy) {
+        Optional<String> mostRecentYear = references.stream()
+                .map(Reference::year)
+                .filter(year -> year != null && !year.isEmpty() && !year.equals("0"))
+                .max(Comparator.naturalOrder());
+
+        if (mostRecentYear.isPresent()) {
+            return mostRecentYear.get();
+        }
+
+        int year;
+        if (therapy != null) {
+            year = !entryDate.isBefore(therapy.createDate()) ? entryDate.getYear() : therapy.createDate().getYear();
```

Comment:
I would reverse the evaluation here (`entryDate.isBefore(therapy.createDate) ?`). 

Also, you could inline this (`int year = `) though that may make reading the code a bit harder, so no strong opinion.

Reply:
My suggestion wasn't complete I think :)

Note that `isAfter` is not the same as `!isBefore` when the dates are equal (but in this use-case it doesn't matter. 

A cleaner implementation could be: `Math.max(therapy.createDate.year(), entryDate.getYear)` in case therapy != null?